packages/experience-builder-sdk/src/hooks/useClassName.ts

```diff
@@ -0,0 +1,61 @@
+import { useLayoutEffect, useState } from 'react';
+
+import {
+  buildCfStyles,
+  buildStyleTag,
+  isEmptyStructureWithRelativeHeight,
+} from '@contentful/experiences-core';
+import { ComponentTreeNode } from '@contentful/experiences-core/types';
+import { EMPTY_CONTAINER_HEIGHT } from '@contentful/experiences-core/constants';
+
+/**
+ * This hook can generate className and inject styles on a client side as a <style> tag
+ * or it derives the className set on the server side
+ *
+ * @param node: the componenet node for which the styles will be injected
+ * @param props: the props of the component, represented by the node
+ * @returns the className that was eitner generated on the client side or derived from the value, set on server side
+ */
+export const useClassName = ({
+  node,
+  props,
+}: {
+  node: ComponentTreeNode;
+  props: Record<string, any>;
+}) => {
+  const [className, setClassName] = useState<string>(props.cfSsrClassName ?? '');
```

Comment:
I don't think you need a useState/useEffect here. This could be a derived `useMemo` or am I missing something here?

Reply:
Maybe the bit that you have missed is that a className can be set from the `useLayoutEffect` [below](https://github.com/contentful/experience-builder/pull/600/files#diff-05c7e60264f1b70aebdc0607d81cc8c661cfc18d607fce7048d623c9401ed7b3R55)

I wouldn't convert the entire thing into `useMemo` because `useLayoutEffect` performs DOM manipulation by creating a `style` tag and adding to the head of the page.